### DISCUSSION TITLE
Error on Install

### DISCUSSION THREAD

**Discussion by osseonews** [vendurehq/vendure#659] [General]
Just tried the basic install and when trying to run `npm run start` after the set up, the server throws the following error:

`node_modules/@vendure/email-plugin/lib/src/types.d.ts(4,28): error TS7016: Could not find a declaration file for module 'nodemailer/lib/mailer'. '/opt/build/repo/node_modules/nodemailer/lib/mailer/index.js' implicitly has an 'any' type.`

Any idea how to solve this?

**Comment by michaelbromley** [comment]
Hi, can you provide a bit more information on how to reproduce. I just tried installing with @vendure/create 0.18.2 and did not see the error:

```
Michael@XPS15 MINGW64 /c/Development/temp
$ npx @vendure/create@0.18.2 issue-659
npx: installed 98 in 23.7s

Welcome to @vendure/create v0.18.2!

Let's configure a new Vendure project. First a few questions:

√ Which database are you using? » SQLite
√ Which programming language will you be using? » TypeScript
√ Populate with some sample product data? ... no / yes
√ What identifier do you want to use for the superadmin user? ... superadmin
√ What password do you want to use for the superadmin user? ... superadmin

Setting up your new Vendure project in C:\Development\temp\issue-659
This may take a few minutes...

  √ Installing dependencies
  √ Generating app scaffold
  √ Initializing server

Success! Created a new Vendure server at C:\Development\temp\issue-659

We suggest that you start by typing:

    cd issue-659
    yarn start

Happy hacking!

Michael@XPS15 MINGW64 /c/Development/temp
$ cd issue-659/

Michael@XPS15 MINGW64 /c/Development/temp/issue-659
$ yarn start
yarn run v1.19.1
$ concurrently yarn:run:*
$ ts-node ./src/index.ts
$ ts-node ./src/index-worker.ts
[run:worker] info 1/18/21, 09:46 - [Vendure Worker] Bootstrapping Vendure Worker (pid: 14496)...
[run:server] info 1/18/21, 09:46 - [Vendure Server] Bootstrapping Vendure Server (pid: 16720)...
[run:worker] info 1/18/21, 09:46 - [NestMicroservice] Nest microservice successfully started
[run:worker] info 1/18/21, 09:46 - [Vendure Worker] Vendure Worker started with TCP transport at localhost:3020
[run:server] info 1/18/21, 09:46 - [RoutesResolver] HealthController {/health}:
[run:server] info 1/18/21, 09:46 - [RouterExplorer] Mapped {/health, GET} route
[run:server] info 1/18/21, 09:46 - [AssetServerPlugin] Asset server listening on "http://localhost:3001"
[run:server] info 1/18/21, 09:46 - [NestApplication] Nest application successfully started
[run:server] info 1/18/21, 09:46 - [Vendure Server] =====================================================================
[run:server] info 1/18/21, 09:46 - [Vendure Server]           Vendure server (v0.18.2) now running on port 3000
[run:server] info 1/18/21, 09:46 - [Vendure Server] ---------------------------------------------------------------------
[run:server] info 1/18/21, 09:46 - [Vendure Server] Shop API:     http://localhost:3000/shop-api
[run:server] info 1/18/21, 09:46 - [Vendure Server] Admin API:    http://localhost:3000/admin-api
[run:server] info 1/18/21, 09:46 - [Vendure Server] Asset Server: http://localhost:3000/assets/ -> http://localhost:3001
[run:server] info 1/18/21, 09:46 - [Vendure Server] Dev Mailbox:  http://localhost:3000/mailbox/ -> http://localhost:3003
[run:server] info 1/18/21, 09:46 - [Vendure Server] Admin UI:     http://localhost:3000/admin/ -> http://localhost:3002
[run:server] info 1/18/21, 09:46 - [Vendure Server] =====================================================================
```

**Comment by osseonews** [comment]
Thanks. This was not a local install. We were trying to do the install on Digital Ocean's App platform via a repo we created. We were using `npm run start` to run the app. 

Ultimately, we got past this error by manually installing the email plugin and also `npm install @types/nodemailer`

But, then DO complained they couldn't connect to SQL Lite. So we installed another package for this that their debugger recommended. Still couldnt connect. 

We then switched over to Postgres and also couldn't connect (I forgot the exact errors). After we installed `npm install pg --save` we got past that error. But unfortunately, DO couldn't run postgres either. This appears to be a bug with TypeORM, where the set up is complicated with databases hosted on Heroku (which DO appears to be using in the background). Something to do with SSL configuration.  Just gave up eventually.

Another issue is that all of these platforms expect the app to be running on port: 8080, not 3000. We changed the port in the vendure config settings, but that didn't help.

We will try to a Droplet + Managed Database next. Maybe that setup will work. 

Perhaps consider setting up a One-Click Image on DO?

**Reply by michaelbromley** [reply]
OK. If you want to provide a step-by-step list of actions you took to attempt to get this running on DO, then I might have more of an idea of why you have been running into these issues. Currently I don't have enough information to work with.

> Perhaps consider setting up a One-Click Image on DO?

Yes, I might look into this in the future.

**Comment by osseonews** [comment]
OK, thanks. We'll try to use a regular DO Node.js droplet first and a managed database. I think that will work better. It could just be that the DO app platform is not up to par yet (just launched).
